src/components/e-table.vue

```diff
@@ -0,0 +1,294 @@
+   * by using an identical named slot binding ('v-slot:\<slotName\>') to the component.
+   */
+  export default {
+    name: 'e-table',
+    status: 0, // TODO: remove when component was prepared for current project.
+
+    // components: {},
+    // mixins: [],
+
+    props: {
+      /**
+       * Array of data objects to render in table.
+       *
+       * @property {Boolean} [disabled = false] - Disables the interaction with the current row.
+       */
+      items: {
+        type: Array,
+        required: true,
+      },
+
+      /**
+       * Array of column definition objects.
+       *
+       * @property {String} title - Column title.
+       * @property {String} [key] - Column identifier on the rows.
+       * @property {String} [titleHidden = false] - Title visibility state.
+       * @property {String} [slotName] - Dynamic slot name for optional cell templates.
+       * @property {String} [align = 'left'] - A flag to set the column alignment (default: 'left', options: 'center', 'right').
+       * @property {Boolean} [sortable = false] - A flag that specifies whether the column is sortable or not.
+       * @property {Function} [sort = (a, b) => a > b] - A custom sort function that might be passed for each column.
+       * @property {Function} [onClick] - Allows to define a click handler for each cell.
+       *
+       */
+      columns: {
+        type: Array,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        /**
+         * @type {Object} The currently selected 'column' to be sorted.
+         */
+        sortBy: null,
+
+        /**
+         * @type {Boolean} Bool which stores the 'sort-direction', default is ascending (true).
+         */
+        sortAscending: null,
+      };
+    },
+
+    computed: {
+      /**
+       * Returns a sorted and modified copy of the table-items.
+       *
+       * @returns {Array}
+       */
+      itemsSorted() {
+        const items = [...this.items]; // Creates a shallow clone to avoid mutating the prop
+
+        if (this.sortBy) {
+          const sort = typeof this.sortBy.sort === 'function'
+            ? this.sortBy.sort
+            : (a, b) => a[this.sortBy.key].localeCompare(b[this.sortBy.key]);
+
+          items.sort(sort);
+        }
+
+        if (this.sortAscending === false) {
+          items.reverse();
+        }
+
+        return items;
+      },
+    },
+    // watch: {},
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      /**
+       * Will set the sort-parameters.
+       *
+       * @param {Object} column - Object which holds the information about the current column.
+       */
+      onClickSort(column) {
+        if (this.sortBy === column) {
+          if (this.sortAscending === null) {
```

Comment:
- [ ] I think it would be a bit easier to read if the sortDirection is a string with the possible values of 'asc', 'desc', 'none'. What do you think about that?